zipline/pipeline/slice.py

```diff
@@ -0,0 +1,55 @@
+
+from numpy import searchsorted
+
+
+class Slice(object):
+    """
+    Mixin for extracting a single column of a Term's output.
+
+    Parameters
+    ----------
+    term : zipline.pipeline.term.Term
+        The term from which to extract a column of data.
+    asset : zipline.assets.Asset
+        The asset corresponding to the column of `term` to be extracted.
+
+    Notes
+    -----
+    Users should rarely construct instances of `Slice` directly. Instead, they
+    should construct instances via indexing, e.g. `MyFactor()[Asset(24)]`.
+    """
+
+    def __new__(cls, term, asset):
+        return super(Slice, cls).__new__(
+            cls,
+            asset=asset,
+            inputs=[term],
+            window_length=0,
+            mask=term.mask,
+            dtype=term.dtype,
+            missing_value=term.missing_value,
+            window_safe=term.window_safe,
+            ndim=1,
+        )
+
+    def __repr__(self):
+        return "{type}({parent_term}, column={asset})".format(
+            type=type(self).__name__,
+            parent_term=type(self.inputs[0]).__name__,
+            asset=self._asset,
+        )
+
```

Comment:
I think that rather than using a Mixin here, we might just want to have Slice be a direct subclass of `ComputableTerm`.

Rationale:

- If we're using it as a mixin, then we have to have separate `FactorSlice`, `FilterSlice` and `ClassifierSlice` types.  If it's just a computable term, then we only have one class.
- Using slice as a mixin means that `FactorSlice` and friends inherit all the methods from `Factor`, but most of them don't work on Slice.